README.md

```diff
@@ -1,3 +1,36 @@
+https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/an-slo-for-baremetal.md for details about this operator.
+
+What does it do?
+
+Cluster-baremetal-operator (CBO) is designed to be an OpenShfit Operator that is responsible for deploying all components required to provision Baremetal servers into worker nodes that join an OpenShift cluster. 
+The components that have knowledge of how to connect and boot a Baremetal Server are encapsulated in an upstream K8s project called metal3.io. The CBO is responsible for making sure that the metal3 deployment consisting of the baremetal-operator (BMO) and Ironic containers is always running on one of the mater nodes within the OpenShift cluster.
+The CBO is also responsible for listening to OpenShift updates to resources that it is watching and take appropriate action. The CBO reports on its own state via the “baremetal” clusterOperator resources as is required by every OpenShift operator.
+The CBO reads, validates and passes information provided in the Provisioning Config Resource (CR) and passes this information to the metal3 deployment. It also creates Secrets that containers within the metal3 deployment use to communicate with each other. Currently, only one copy of the Provisioning CR exists per OpenShift Cluster so all worker nodes would be provisioned using the same configuration.
+
+
+When is CBO active?
+
+CBO runs on all platform types supported by OpenShift but will perform its above mentioned tasks only when the platform type is BareMetal. The “baremetal” ClusterOperator displays the current state of the CBO when running on a BareMetal platform and as “Disabled” in all Platform types. 
+When the CBO is running on the BareMetal platform, it manages the metal3 deployment and will continue communicating its state using the “baremetal” ClusterOperator.
+CBO is considered a second level operator (SLO) in OpenShift parlance. What that means is that another OpenShift operator is responsible for deploying CBO. In this case, the Cluster Version Operator (CVO) is responsible for deploying its SLOs at a specific run level that is coded into the manifests of that operator.
+The OpenShift Installer is responsible for deploying the control plane but does not wait for CVO to complete deployment of the CBO. The CVO, also running on the control plane, is completely responsible for running the CBO on one master node at a time. The worker deployment is completely handled by metal3 which in turn deployed by CBO as mentioned earlier.
+
+What are its inputs?
+
+The CBO needs information about the network to which the baremetal servers are connected and where it can find the image required to boot the servers. This information is provided by the Provisioning CR. CBO watches this resource and passes this information to the metal3 deployment after validating its contents.
+The Provisioning CR with details about the config items can be found here. 
```

Comment:
I think here we can document the provisioning CR and talk about network settings.  I'll start working on that.